web/templates/includes/panel.php

```diff
@@ -166,6 +166,20 @@ class="top-bar-menu-link u-hide-tablet"
 								<?php } ?>
 							<?php } ?>
 
+							<!-- Web Terminal -->
+							<?php if (isset($_SESSION["WEB_TERMINAL"]) && !empty($_SESSION["WEB_TERMINAL"]) && $_SESSION["WEB_TERMINAL"] == "true") { ?>
+								<?php if ($_SESSION["userContext"] === "admin" &&  $_SESSION["look"] === "admin" && $_SESSION["POLICY_SYSTEM_PROTECTED_ADMIN"] == "yes") { ?>
+									<!-- Hide web terminal when impersonating admin -->
+								<?php } else { ?>
+									<li class="top-bar-menu-item">
+										<a title="<?= _("Web terminal") ?>" class="top-bar-menu-link <?php if ($TAB == 'TERMINAL') echo 'active' ?>" href="/list/terminal/">
```

Comment:
We probably should limit web shell access to only users where ssh access is enabled

Reply:
I think the terminal works even without SSH access enabled because it doesn't use SSH. It creates a pty running as the user.

Reply:
User set to 'nologin'  can currently not run it anyway so we should probably hide it anyway it is also very useful give certain clients no access...

Reply:
Yes probally useful

Reply:
Should work now. Also added a redirect in case the user somehow goes on the page.